### DISCUSSION TITLE
How to simulates the addition of tuned mass dampers (TMDs) in the nacelle for structural control

### DISCUSSION THREAD

**Discussion by BluewaterHxb** [OpenFAST/openfast#1540] [Q&A] [answered]
How to add the TMD Module in FAST? I want some help to slove this vibration control issue.

**Comment by jjonkman** [comment, marked as answer]
Dear @BluewaterHxb,

TMDs are available in OpenFAST as part of the Structural Control (StC) submodel of ServoDyn, which is documented here: https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_index.html.

I would suggest using the following OpenFAST r-test as a reference: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/StC_test_OC4Semi_Linear_Nac.

Best regards,

**Comment by BluewaterHxb** [comment]
Thank you for your nice response!



---- Replied Message ----
| From | ***@***.***> |
| Date | 04/21/2023 00:14 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [OpenFAST/openfast] How to simulates the addition of tuned mass dampers (TMDs) in the nacelle for structural control (Discussion #1540) |

Dear @BluewaterHxb,

TMDs are available in OpenFAST as part of the Structural Control (StC) submodel of ServoDyn, which is documented here: https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_index.html.

I would suggest using the following OpenFAST r-test as a reference: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/StC_test_OC4Semi_Linear_Nac.

Best regards,
